_this.resolveReadyUpPromise is not a function

```
 test readied up                                                                                                       
 /home/creaturechess/server/src/shared/game/player/player.ts:393                                                       
         this.resolveReadyUpPromise();                                                                                 
              ^                                                                                                        
 TypeError: _this.resolveReadyUpPromise is not a function                                                              
     at Socket.Player.readyUp (/home/creaturechess/server/src/shared/game/player/player.ts:393:14)                     
     at Socket.emit (events.js:198:13)                                                                                 
     at /home/creaturechess/server/node_modules/socket.io/lib/socket.js:528:12                                         
     at process._tickCallback (internal/process/next_tick.js:61:11)                                                    
 npm ERR! code ELIFECYCLE                                                                                              
 npm ERR! errno 1                                                                                                      
 npm ERR! creature-chess@1.0.0 start:server: `npm run build:server && node -r ts-node/register -r tsconfig-paths/regist
 npm ERR! Exit status 1                                                                                                
 npm ERR!                                                                                                              
 npm ERR! Failed at the creature-chess@1.0.0 start:server script.                                                      
 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.                    
 npm ERR! A complete log of this run can be found in:                                                                  
 npm ERR!     /root/.npm/_logs/2019-08-30T16_10_37_753Z-debug.log                                                      
 creaturechess-server.service: Main process exited, code=exited, status=1/FAILURE                                      
 creaturechess-server.service: Failed with result 'exit-code'.                                                         
```

--- a/src/shared/game/player/player.ts
+++ b/src/shared/game/player/player.ts
@@ -1,4 +1,5 @@
 import uuid = require("uuid/v4");
+import pDefer = require("p-defer");
 import { clonePiece, createPieceFromCard } from "../../piece-utils";
 import { MovePiecePacket } from "../../packet-opcodes";
 import { Match } from "../../match/match";
@@ -54,8 +55,7 @@ export abstract class Player {
     private deck: CardDeck;
     private gamePhase: GamePhase = GamePhase.WAITING;
 
-    private readyUpPromise: Promise<void> = null;
-    private resolveReadyUpPromise: () => void = null;
+    private readyUpDeferred: pDefer.DeferredPromise<void>;
 
     private board = new PlayerBoard();
     private turnCount: number;
@@ -121,9 +121,7 @@ export abstract class Player {
     public async enterPreparingPhase(round: number) {
         this.gamePhase = GamePhase.PREPARING;
 
-        this.readyUpPromise = new Promise(resolve => {
-            this.resolveReadyUpPromise = resolve;
-        });
+        this.readyUpDeferred = pDefer();
 
         this.rerollCards();
         this.board.unlockEvolution();
@@ -134,7 +132,7 @@ export abstract class Player {
             return;
         }
 
-        await this.readyUpPromise;
+        await this.readyUpDeferred.promise;
     }
 
     public reset() {
@@ -143,8 +141,7 @@ export abstract class Player {
 
     public enterReadyPhase(turnSimulator: TurnSimulator, opponentProvider: OpponentProvider) {
         this.gamePhase = GamePhase.READY;
-        this.readyUpPromise = null;
-        this.resolveReadyUpPromise = null;
+        this.readyUpDeferred = null;
         this.ready = false;
 
         if (this.isAlive()) {
@@ -382,7 +379,7 @@ export abstract class Player {
     }
 
     protected readyUp = () => {
-        if (this.ready) {
+        if (this.ready || this.readyUpDeferred === null) {
             return;
         }
 
@@ -390,7 +387,7 @@ export abstract class Player {
 
         this.setReady(true);
 
-        this.resolveReadyUpPromise();
+        this.readyUpDeferred.resolve();
     }
 
     protected movePieceToBench = (packet: MovePiecePacket) => {

